File: src/main/java/mediathek/gui/duplicates/statistics/DuplicateStatisticsTableFormat.java
package mediathek.gui.duplicates.statistics;

import ca.odell.glazedlists.gui.AdvancedTableFormat;
import mediathek.gui.duplicates.FilmStatistics;

import java.util.Comparator;

public class DuplicateStatisticsTableFormat implements AdvancedTableFormat<FilmStatistics> {

    @Override
    public int getColumnCount() {
        return 2;
    }

    @Override
    public String getColumnName(int column) {
        return switch (column) {
            case 0 -> "Sender";
            case 1 -> "Anzahl";
            default -> null;
        };
    }

    @Override
    public Object getColumnValue(FilmStatistics stats, int column) {
        return switch (column) {
            case 0 -> stats.sender();
            case 1 -> stats.count();
            default -> null;
        };
    }

    @Override
    public Class getColumnClass(int column) {
        return switch (column) {
            case 0 -> String.class;
            case 1 -> Long.class;
            default -> null;
        };
    }

    @Override
    public Comparator getColumnComparator(int column) {
        switch (column) {
            case 0 -> {
                return (Comparator<String>) String::compareTo;
            }
            case 1 -> {
                return (Comparator<Long>) Long::compareTo;
            }
        }
        return null;
    }
}

File: src/main/kotlin/mediathek/gui/duplicates/statistics/DuplicateStatisticsTableFormat.kt
package mediathek.gui.duplicates.statistics

import ca.odell.glazedlists.gui.AdvancedTableFormat
import mediathek.gui.duplicates.FilmStatistics

class DuplicateStatisticsTableFormat : AdvancedTableFormat<FilmStatistics> {
    override fun getColumnCount(): Int = 2

    override fun getColumnName(column: Int): String = when (column) {
        0 -> "Sender"
        1 -> "Anzahl"
        else -> throw IndexOutOfBoundsException(column)
    }

    override fun getColumnValue(baseObject: FilmStatistics, column: Int): Any? = when (column) {
        0 -> baseObject.sender
        1 -> baseObject.count
        else -> null
    }

    override fun getColumnClass(column: Int): Class<*> = when (column) {
        0 -> String::class.java
        1 -> Long::class.java
        else -> throw IndexOutOfBoundsException(column)
    }

    override fun getColumnComparator(column: Int): Comparator<*>? = when (column) {
        0 -> naturalOrder<String>()
        1 -> naturalOrder<Long>()
        else -> null
    }
}
